File: src/api/java/model/organization/entity/Role.java
package model.organization.entity;

import model.organization.Organization;
import model.organization.function.Goodness;
import model.organization.id.Identifiable;

/**
 * The {@linkplain Role} interface defines a role entity of an {@linkplain Organization}.
 *
 * @author Christopher Zhong
 * @see Goodness
 * @since 3.4
 */
public interface Role extends Identifiable<Role> {}

File: chazm-api/src/main/kotlin/runtimemodels/chazm/api/entity/Role.kt
package runtimemodels.chazm.api.entity

import runtimemodels.chazm.api.function.Goodness
import runtimemodels.chazm.api.id.Identifiable
import runtimemodels.chazm.api.id.RoleId
import runtimemodels.chazm.api.organization.Organization

/**
 * The [Role] interface defines a role entity of an [Organization].
 *
 * @author Christopher Zhong
 * @see Goodness
 *
 * @since 3.4
 */
interface Role : Identifiable<Role, RoleId>
